README.background-tasks.md

```diff
@@ -0,0 +1,119 @@
+# ACSF Tools
+
+## Summary:
+
+Background tasks are designed to execute long running drush commands in the
+background by using multiple concurrent cron jobs. One use case is to offload
+long running "light" tasks from deployment to post deployment to achieve higher
+concurrency.
+
+## Installation and Configuration:
+
+#### Install
+
+0. Fist you are going to need to request the approval to use this in your own ACSF account, and a piece of code that will allow you to do it. Contact PS or your TAM for that.
+
+In particular you'll need acsf_large_scale_cron.rb
+
+
+1. Import acsf_tools with background tasks:
+
+```
+composer require acquia/acsf-tools:dev-background-tasks-10.x-dev --no-update
+composer update acquia/acsf-tools --no-dev
+```
+
+or apply the patch
+
+```
+https://github.com/acquia/acsf-tools/pull/83.patch
+```
+
+2. Add following to db-update.sh after database is updated (eg, after drush updb):
+
+```
+# ############# POST DEPLOYMENT TASKS RELATED ############################
+# Custom argument. Currently used for triggering usage of post deployment tasks.
+# To trigger post deployment tasks execution using cron jobs, pass "usecronbackgroundtasks"
+custom_argument="$5"
+
+docroot="/var/www/html/$sitegroup.$env/docroot"
+DRUSH_CMD="drush9 --root=$docroot --uri=https://$domain"
+
+# Fetch if post deployment tasks are pending.
+CMD1="$(DRUSH_PATHS_CACHE_DIRECTORY="$cache_dir" $DRUSH_CMD sget cohesion_import_pending  --format=json | grep cohesion_import_pending | grep -o '.$')"
```

Comment:
Do we really need to mention Cohesion here?

Reply:
This is an example use case. I think it works well for illustration purposes.